### PR TITLE
Atropos outputs

### PR DESCRIPTION
**Pull request by tanaes**
Hey Semar, 

I'm going through and looking at outputs. Going to put comments on this PR.

### CODE REVIEW

File: qp_shotgun/qc_trim/qc_trim.py [hunk 1]
```diff
@@ -161,14 +161,15 @@ def generate_qc_trim_commands(forward_seqs, reverse_seqs, map_file,
     """
     # we match filenames, samples, and run prefixes
     samples = make_read_pairs_per_sample(forward_seqs, reverse_seqs, map_file)
-	#threads = 4
```
Comment (tanaes) [inline qp_shotgun/qc_trim/qc_trim.py:164]: Make sure to convert tabs to spaces!

File: qp_shotgun/qc_trim/qc_trim.py [hunk 2]
```diff
@@ -161,14 +161,15 @@ def generate_qc_trim_commands(forward_seqs, reverse_seqs, map_file,
...
+    #threads = 4
     cmds = []
 
     param_string = _format_qc_trim_params(parameters)
     for run_prefix, sample, f_fp, r_fp in samples:
         r_fp_str = ' --input "%s"' % r_fp if r_fp is not None else ""
```
Comment (tanaes) [inline qp_shotgun/qc_trim/qc_trim.py]: Note that the `-o` and `-p` flags in Atropos indicate the output file names. We should make them different from the input file names, perhaps by adding `_trimmed` to the filename prior to the extension?

File: qp_shotgun/qc_trim/qc_trim.py [hunk 3]
```diff
@@ -161,14 +161,14 @@ def generate_qc_trim_commands(forward_seqs, reverse_seqs, map_file,
     """
     # we match filenames, samples, and run prefixes
     samples = make_read_pairs_per_sample(forward_seqs, reverse_seqs, map_file)
-	#threads = 4
+    #threads = 4
     cmds = []
 
     param_string = _format_qc_trim_params(parameters)
     for run_prefix, sample, f_fp, r_fp in samples:
-        r_fp_str = ' --input "%s"' % r_fp if r_fp is not None else ""
```
Comment (tanaes) [inline qp_shotgun/qc_trim/qc_trim.py:169]: Do we want to support forward-only reads for now? Will need bigger changes to Atropos command string generation -- maybe easier to start with only paired-end case for now.

File: qp_shotgun/qc_trim/qc_trim.py [hunk 4]
```diff
@@ -186,24 +186,11 @@ def _run_commands(qclient, job_id, commands, msg):
     return True, ""
 
 
-def _gzip_file(path):
```
Comment (tanaes) [inline qp_shotgun/qc_trim/qc_trim.py:189]: Atropos will automatically gzip files that are input as gzipped files, obviating need for this subroutine. 

I do think that this requires gzipping all input fastqs by default further up the pipeline though... We should see if this is already being done.

File: qp_shotgun/qc_trim/qc_trim.py [hunk 5]
```diff
@@ -186,24 +186,11 @@ def _run_commands(qclient, job_id, commands, msg):
...
-    return gz_path
-
-
 def _per_sample_ainfo(out_dir, samples, fwd_and_rev=False):
     files = []
     missing_files = []
 
-    if fwd_and_rev:
-        suffixes = ['%s_paired_1.fastq', '%s_paired_2.fastq',
-                    '%s_unmatched_1.fastq', '%s_unmatched_2.fastq']
```
Comment (tanaes) [inline qp_shotgun/qc_trim/qc_trim.py:204]: Atropos doesn't output unmatched read pairs given our parameters -- only need to keep the paired case (and we should probably also rename ... e.g. `sample1_R1_trimmed.fastq.gz`)

File: qp_shotgun/qc_trim/qc_trim.py [hunk 6]
```diff
@@ -219,14 +206,6 @@ def _per_sample_ainfo(out_dir, samples, fwd_and_rev=False):
         # was kept after quality control and filtering for host data
         raise ValueError("No sequences left after running KneadData")
 
-    # Generate the missing files
```
Comment (tanaes) [inline qp_shotgun/qc_trim/qc_trim.py:222]: We shouldn't need to generate missing files now... but want to check in on the edge case of very low read # input files to test this.

File: qp_shotgun/qc_trim/tests/test_qc_trim.py [hunk 7]
```diff
@@ -180,9 +181,9 @@ def test_generate_qc_trim_analysis_commands_forward_reverse(self):
         self._clean_up_files.append(fp)
 
         exp_cmd = [
-			'atropos --threads 4 --A GATCGGAAGAGCGTCGTGTAGGGAAAGGAGTGT --adapter GATCGGAAGAGCACACGTCTGAACTCCAGTCAC --max-n 80 --minimum-length 80 --nextseq-trim False --pair-filter any --quality-cutoff 15 --trim-n True -o output/fastq/s1.fastq -p output/fastq/s1.R2.fastq -pe1 fastq/s1.fastq -pe2 fastq/s1.R2.fastq',
```
Comment (tanaes) [inline qp_shotgun/qc_trim/tests/test_qc_trim.py:183]: Note that I'm changing up the output naming here... We want to modify f_fp and r_fp to create unique output filenames, I think.

**Review by tanaes — COMMENTED** [review]
Hey @semarpetrus, i've gone through and done a quick review focusing out Atropos-specific outputs. I'm sorry I don't have time this morning to implement these, but is this clear enough to get going? Please hit me back with any comments or questions.
